bin/radical-pilot-service-wrapper

```diff
@@ -0,0 +1,297 @@
+#
+# This script is a wrapper around an actual executable.  It's purpose is,
+# depening on the mode of operation, to either watch the executable's stdout for
+# some pattern, or to watch an output file created by the executable for that
+# pattern.  If the pattern is found, the matching data is extracted and sent to
+# *this* script's stdout.  If the executable returns a non-zero exit code, or if
+# a timeout is reached, the script will also return a non-zero exit code.  If
+# The script returns a zero exit code but a pattern is provided but not found,
+# the script will also return a non-zero exit code.
+#
+# The script is called with the following arguments:
+#
+#   -m mode   : data stream to search through, either `stdout` (default),
+#               `stderr`, or a file name
+#   -p pattern: regex pattern to search for
+#   -o output : file name to store the executable's stdout
+#   -e error  : file name to store the executable's stderr
+#   -t timeout: time in seconds after which the script will terminate if pattern
+#               is not found
+#   --        : marks the end of the script's arguments and the start of the
+#               executable's name and arguments
+
+
+class ServiceWrapper(object):
+
+    def __init__(self):
+
+        parser = argparse.ArgumentParser(description='')
+
+        parser.add_argument('-c', '--command',  dest='command')
+        parser.add_argument('-r', '--registry', dest='reg_url')
+        parser.add_argument('-u', '--uid',      dest='uid')
+        parser.add_argument('-m', '--mode',     dest='mode')
+        parser.add_argument('-p', '--pattern',  dest='pattern')
+        parser.add_argument('-o', '--output',   dest='output')
+        parser.add_argument('-e', '--error',    dest='error')
+        parser.add_argument('-v', '--verbose',  dest='verbose', action='store_true')
+        parser.add_argument('-t', '--timeout',  dest='timeout')
+
+        args = parser.parse_args()
+        self._command = args.command or None
+        self._reg_url = args.reg_url or os.environ.get('RP_REGISTRY_ADDRESS')
+        self._uid     = args.uid     or os.environ.get('RP_TASK_ID')
+        self._mode    = args.mode    or 'stdout'
+        self._pattern = args.pattern or None
+        self._output  = args.output  or '%s.stdout' % self._uid
+        self._error   = args.error   or '%s.stderr' % self._uid
+        self._verbose = args.verbose or False
+        self._timeout = args.timeout or 0
+
+        self._timeout = int(self._timeout) if self._timeout else 0
+
+        if not self._command: raise ValueError('no command given')
+        if not self._uid    : raise ValueError('no uid given')
+        if not self._reg_url: raise ValueError('RP_REGISTRY_ADDRESS not set')
+
+        self._sbox = os.environ.get('RP_TASK_SANDBOX', './')
+        self._log  = ru.Logger('radical.pilot.service_wrapper',
+                               targets='%s/%s.log' % (self._sbox, self._uid))
```

Comment:
within the short run (~30sec), this log hasn't been created

Reply:
I see the logfile reliably being created in the service task sandbox.  Please open a new ticket if this persists for your use case.